nutanix/data_source_nutanix_era_profiles.go

```diff
@@ -0,0 +1,424 @@
+		return diag.FromErr(e)
+	}
+
+	uuid, er := uuid.GenerateUUID()
+
+	if er != nil {
+		return diag.Errorf("Error generating UUID for era clusters: %+v", err)
+	}
+	d.SetId(uuid)
+	return nil
+}
+
+func flattenVersions(erv []*Era.Versions) []map[string]interface{} {
+	if len(erv) > 0 {
+		res := make([]map[string]interface{}, len(erv))
+
+		for k, v := range erv {
+			ents := make(map[string]interface{})
+			ents["id"] = v.ID
+			ents["name"] = v.Name
+			ents["description"] = v.Description
+
+			ents["status"] = v.Status
+			ents["owner"] = v.Owner
+			ents["engine_type"] = v.Enginetype
+
+			ents["type"] = v.Type
+			ents["topology"] = v.Topology
+			ents["db_version"] = v.Dbversion
+
+			ents["system_profile"] = v.Systemprofile
+			ents["version"] = v.Version
+			ents["profile_id"] = v.Profileid
+
+			ents["published"] = v.Published
+			ents["deprecated"] = v.Deprecated
+
+			ents["properties"] = flattenProperties(v.Properties)
+			ents["properties_map"] = flattenPropertiesMap(v.Propertiesmap)
+			ents["version_cluster_association"] = flattenClusterAssociation(v.VersionClusterAssociation)
+			res[k] = ents
+		}
+		return res
+	}
+	return nil
+}
+
+func flattenProperties(erp []*Era.Properties) []map[string]interface{} {
+	if len(erp) > 0 {
+		res := make([]map[string]interface{}, len(erp))
+
+		for k, v := range erp {
+			ents := make(map[string]interface{})
+			ents["name"] = v.Name
+			ents["value"] = v.Value
+			ents["secure"] = v.Secure
+			res[k] = ents
+		}
+		return res
+	}
+	return nil
+}
+
+func flattenProfilesResponse(erp *Era.ProfileListResponse) []map[string]interface{} {
+	if erp != nil {
+		lst := []map[string]interface{}{}
+		for _, v := range *erp {
+			d := map[string]interface{}{}
+			d["id"] = v.ID
+			d["name"] = v.Name
+			d["description"] = v.Description
+			d["status"] = v.Status
+			d["owner"] = v.Owner
+			d["engine_type"] = v.Enginetype
+			d["type"] = v.Type
+			d["topology"] = v.Topology
+			d["db_version"] = v.Dbversion
+			d["system_profile"] = v.Systemprofile
+			d["latest_version"] = v.Latestversion
+			d["latest_version_id"] = v.Latestversionid
+			d["versions"] = flattenVersions(v.Versions)
+
+			lst = append(lst, d)
+		}
+		return lst
+	}
+	return nil
+}
+
+func flattenClusterAssociation(erc []*Era.VersionClusterAssociation) []map[string]interface{} {
+	if len(erc) > 0 {
+		res := make([]map[string]interface{}, len(erc))
+
+		for k, v := range erc {
+			ercs := map[string]interface{}{}
+
+			ercs["nx_cluster_id"] = v.NxClusterID
+			ercs["date_created"] = v.DateCreated
+			ercs["date_modified"] = v.DateModified
+			ercs["owner_id"] = v.OwnerID
+			ercs["status"] = v.Status
+			ercs["profile_version_id"] = v.ProfileVersionID
+			ercs["properties"] = flattenProperties(v.Properties)
+			ercs["optimized_for_provisioning"] = v.OptimizedForProvisioning
+
+			res[k] = ercs
+		}
+		return res
+	}
+	return nil
+}
+
+func flattenPropertiesMap(prm map[string]interface{}) map[string]interface{} {
```

Comment:
Do we need this seperate function ? as its just a small check

Reply:
updated the code. Removed the func.